app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -78,8 +79,12 @@ def create
           return
         end
         datacenter = change_datacenter_to_uuid(compute_resource_params[:datacenter])
-        @compute_resource.datacenter = datacenter if datacenter.present?
-
+        if compute_resource_params["provider"].downcase == "ovirt"
+          # create a CR with DC uuid
+          @compute_resource = ComputeResource.new_provider(compute_resource_params.merge!(datacenter: datacenter))
+        else
+          @compute_resource.datacenter = datacenter if datacenter.present?
```

Comment:
Is this actually needed for non-ovirt resources? considering we already passed the datacenter parameter previously and `datacenter` will be `nil` now.
How about we change the way this function is structured to something like:
```ruby
      def create
        begin
          if compute_resource_params["provider"].downcase == "ovirt" && !compute_resource_params[:datacenter].is_uuid?
              @compute_resource = ComputeResource.new_provider(compute_resource_params.except(:datacenter))
              compute_resource_params[:datacenter] = change_datacenter_to_uuid(compute_resource_params[:datacenter])
          end
          @compute_resource = ComputeResource.new_provider(compute_resource_params)
        rescue Foreman::Exception => e
          render_exception(e, :status => :unprocessable_entity)
          return
        end

        process_response @compute_resource.save
      end
```
maybe we could do the same on update as well, and extract the whole `if` block to a `before_action` that is only triggered for ovirt crs?

Reply:
I wouldn't like to change the UI behavior for 6.8, I'm afraid we don't have enough tests for ovirt, and I'm afraid that something goes wrong for the fourth time. 
the update should not be a problem because the CR already have a connection to the ovirt provider.